lib/base_service.js

```diff
@@ -31,13 +32,39 @@ class BaseService {
         this._logger = null;
         this._metrics = null;
         this._ratelimiter = null;
-        this.serviceReturns = null;
+        this._serviceReturns = null;
+
+        this._uncaughtExceptionHandler = (err) => {
+            this._logUnhandledException(err);
+            // Don't exit right away, allow logger to process message
+            setTimeout(() => {
+                process.exit(1);
+            }, 100);
+        };
+
+        // Avoid recursion if there are bugs in the logging code.
+        this._inLogger = false;
+
+        this._unhandledRejectionHandler = (err) => {
+            if (!this._inLogger) {
+                this._inLogger = true;
+                this._logger.log('fatal/service-runner/unhandled', err);
+                this._inLogger = false;
+            }
+        };
     }
 
     stop() {
         if (this._ratelimiter) {
             this._ratelimiter.stop();
         }
+        // Remove handlers
+        if (this._unhandledRejectionHandler) {
```

Comment:
This will always evaluate to 'true'

Reply:
Removed. This is the result of  code change, I had original set the property to null.